### PR TITLE
[pilot] Pilot update description on upload

### PR DESCRIPTION
**Pull request by tadelv**
Thanks for contributing to _fastlane_! Before you submit your pull request, please make sure to check the following boxes:

- [] Run `bundle exec rspec` from the subdirectory of each tool you modified. Alternatively, run `rake test_all` from the root directory.
- [] Run `bundle exec rubocop -a` to ensure the code style is valid
- [] Read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [] We currently don't accept new actions, please publish a plugin instead, more information in [Plugins.md](https://github.com/fastlane/fastlane/blob/master/fastlane/docs/Plugins.md)

Before submitting a pull request, we appreciate if you create an issue first to discuss the change :+1:

an update to https://github.com/fastlane/fastlane/pull/6631. This is a new version of the previous pull request. The changes are rebased to the current master.

### CODE REVIEW

File: pilot/lib/pilot/build_manager.rb [hunk 1]
```diff
@@ -127,6 +133,16 @@ def wait_for_processing_build
...
           break if builds.count == 0
           latest_build = builds.last
+
+          if latest_build.valid and must_update_build_info
+            # Set the changelog and/or description if necessary
+            if should_update_build_information(options)
+              build.update_build_information!(whats_new: options[:changelog], description: options[:beta_app_description], feedback_email: options[:beta_app_feedback_email])
+              UI.success "Successfully set the changelog and/or description for build"
+            end
+            must_update_build_info = false
```
Comment (milch) [inline pilot/lib/pilot/build_manager.rb]: This doesn't do anything, remove please.

**Review by milch — CHANGES_REQUESTED** [review]
Thanks for your submission! 

Looks good to me, just a small change needed please 🚀

File: pilot/lib/pilot/build_manager.rb (hunk 1 above)
Comment (tadelv) [inline pilot/lib/pilot/build_manager.rb]: I'm not sure I understand. I need to set this flag in order **not** to update build info on every pass of the loop while still waiting for the build to be fully processed. Am I missing something?

Comment (milch) [inline pilot/lib/pilot/build_manager.rb]: Ah, true. I must've blacked out there for a second and missed the loop.

**Review by milch — APPROVED** [review]

File: pilot/lib/pilot/build_manager.rb [hunk 2]
```diff
@@ -63,10 +63,11 @@ def distribute(options, build = nil)
         UI.message("Distributing build #{build.train_version}(#{build.build_version}) from #{build.testing_status} -> External")
       end
 
-      # First, set the changelog and/or description if necessary
-      if options[:changelog].to_s.length > 0 or options[:beta_app_description].to_s.length > 0 or options[:beta_app_feedback_email].to_s.length > 0
-        build.update_build_information!(whats_new: options[:changelog], description: options[:beta_app_description], feedback_email: options[:beta_app_feedback_email])
-        UI.success "Successfully set the changelog and/or description for build"
+      unless config[:update_build_info_on_upload]
+        if should_update_build_information(options)
+          build.update_build_information!(whats_new: options[:changelog], description: options[:beta_app_description], feedback_email: options[:beta_app_feedback_email])
```
Comment (KrauseFx) [inline pilot/lib/pilot/build_manager.rb]: Seems like `build` is undefined here

File: pilot/lib/pilot/build_manager.rb [hunk 3]
```diff
@@ -127,6 +133,16 @@ def wait_for_processing_build
           builds = app.all_processing_builds
           break if builds.count == 0
           latest_build = builds.last
+
+          if latest_build.valid and must_update_build_info
+            # Set the changelog and/or description if necessary
+            if should_update_build_information(options)
+              build.update_build_information!(whats_new: options[:changelog], description: options[:beta_app_description], feedback_email: options[:beta_app_feedback_email])
```
Comment (milch) [inline pilot/lib/pilot/build_manager.rb]: For posterity: This line introduced the issue #7439

**Pull request merged**